mcm/json_layer/request.py

```diff
@@ -1864,6 +1864,64 @@ def test_failure(self, message, what='Submission', rewind=False, with_notificati
 
         self.reload()
 
+    def _attempt_filter_workflows(self, mcm_reqmgr_name_list, stats_workflows):
+        """
+        Attempts to pick the latest injected workflows using the records available
+        in McM. After an injection process finishes, only the latest workflow name
+        is available on the `reqmgr_name` attribute with no content.
+
+        Note this filter process is not idempotent. In case the request has been
+        updated with previous data from Stats2, it is not possible to get the latest
+        workflow and nothing will be filtered. Such behavior is acceptable.
+
+        Args:
+            mcm_reqmgr_name_list (list[str]): List of ReqMgr2 workflow names available for
+                the request in McM.
+            stats_workflows (list[dict]): Stats2 workflow records related to the McM
+                request.
+
+        Returns:
+            list[dict]: Stats2 workflows possibly filtered by the latest injected
+                workflow. This could return the original list if there reference to
+                the latest injected workflow is not available or an empty list if
+                the latest injected workflow was not found in the provided list.
+        """
+        stats_workflows_dict = {
+            stats_workflow.get('RequestName'): stats_workflow
+            for stats_workflow in stats_workflows
+        }
+        workflows_to_consider = [
+            stats_workflows_dict.get(mcm_reqmgr_name)
+            for mcm_reqmgr_name in mcm_reqmgr_name_list
+            if stats_workflows_dict.get(mcm_reqmgr_name)
+        ]
+
+        self.logger.info("Workflows to consider: %s", [doc.get('RequestName') for doc in workflows_to_consider])
+        try:
+            # Pick the time related to the `new` transition
+            reference_time_of_new = min([
+                stats_doc.get('RequestTransition')[0].get('UpdateTime')
+                for stats_doc in workflows_to_consider
+            ])
+        except:
+            # There's no record for the latest injected workflow in Stats2.
+            return []
+
+        stats_reqmgr_name_list = [
+            reqmgr_name
+            for reqmgr_name, content in stats_workflows_dict.iteritems()
+            if content.get('RequestTransition') and
+                len(content['RequestTransition']) > 0 and
```

Comment:
Isn't this redundant? `bool([])` is `False`

Reply:
A bit, just to make clear the attribute is a `list`